### PR TITLE
Fix wrong deploy option for klusterlet

### PR DESCRIPTION
**Pull request by qiujian16**
Signed-off-by: Jian Qiu <jqiu@redhat.com>

### CODE REVIEW

File: operator/v1/types.go [hunk 1]
```diff
@@ -243,13 +249,11 @@ type KlusterletSpec struct {
 	Namespace string `json:"namespace,omitempty"`
 
 	// RegistrationImagePullSpec represents the desired image configuration of registration agent.
-	// +required
-	// +kubebuilder:default=quay.io/open-cluster-management/registration
-	RegistrationImagePullSpec string `json:"registrationImagePullSpec"`
+	// +optional
```
Comment (zhujian7) [inline operator/v1/types.go:253]: Even if we removed the `default` of the `RegistrationImagePullSpec` and `WorkImagePullSpec`, should they remain `required`?

Comment (qiujian16) [reply operator/v1/types.go:253]: it is for backward compatible, I do not think we require this in the previous version

Comment (qiujian16) [reply operator/v1/types.go:253]: will add a comment on what image is used if unspecified.

**Comment by zhujian7** [comment]
LGTM

File: operator/v1/0001_00_operator.open-cluster-management.io_klusterlets.crd.yaml [hunk 2]
```diff
@@ -38,52 +38,10 @@ spec:
               type: object
               default:
```
Comment (zhujian7) [inline operator/v1/0001_00_operator.open-cluster-management.io_klusterlets.crd.yaml]: Still has a `default` value here

File: operator/v1/types.go [hunk 3]
```diff
@@ -269,7 +275,7 @@ type KlusterletSpec struct {
 	// DeployOption contains the options of deploying a klusterlet
 	// +optional
 	// +kubebuilder:default={mode: Default}
```
Comment (zhujian7) [inline operator/v1/types.go]: Should delete this line too.

**Review by zhujian7 — CHANGES_REQUESTED** [review]

**Pull request merged**